test/semantics/select-case.f90

```diff
@@ -0,0 +1,172 @@
+! Test SELECT CASE Constraints: C1145, C1146, C1147, C1148, C1149
+program selectCaseProg
+implicit none
+   ! local variable declaration
+   character :: grade1 = 'B'
+   integer :: grade2 = 3
+   logical :: grade3 = .false.
+   character (len = 10) :: name = 'test'
+   real :: grade4 = 2.0
+   logical, parameter :: grade5 = .false.
+   CHARACTER(KIND=1), PARAMETER    :: ze = 'a', at='b'
+   CHARACTER(KIND=4), PARAMETER :: mh='c'
+
+   select case (grade1)
+      case ('A')
+      print*, "Excellent!"
+      case ('B')
+      case ('C')
+         print*, "Well done"
+      case default
+         print*, "Invalid grade"
+   end select
+
+   select case (grade2)
+      case (1)
+      print*, "Excellent!"
+      case (2)
+      case (3)
+         print*, "Well done"
+      case default
+         print*, "Invalid grade"
+   end select
+
+   select case (grade3)
+      case (.true.)
+       name = 'true'
+      case (.false.)
+        name = 'false'
+   end select
+   print*, "Your grade is ", name
+
+   select case (name)
+      case default
+         print*, "Invalid grade"
+      case ('now')
+      case ('test')
+         print*, "Well done"
+   end select
```

Comment:
Tests are needed for other kinds of `CHARACTER`.

Reply:
I have added check for all 3 character kinds, 1, 2 and 4 in the test case.